tests/streamlit_ui_tests/streamlit.spec.ts

```diff
@@ -376,4 +376,38 @@ test.describe('Keyboard Shortcuts', () => {
     await expect(mito.getByPlaceholder('Find...')).toBeVisible()
     await expect(mito.getByPlaceholder('Replace...')).toBeVisible()
   })
+
+  test('Set Number Format', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await mito.getByTitle('Column1').click();
+
+    await page.keyboard.press('Control+Shift+1');
+    await awaitResponse(page);
+    await expect(mito.getByText('1.00', { exact: true })).toBeVisible();
+
+    await page.keyboard.press('Control+Shift+4');
+    await awaitResponse(page);
+    await expect(mito.getByText('$1.00', { exact: true })).toBeVisible();
+
+    await page.keyboard.press('Control+Shift+5');
+    await awaitResponse(page);
+    await expect(mito.getByText('100.00%', { exact: true })).toBeVisible();
+
+    await page.keyboard.press('Control+Shift+^');
+    await awaitResponse(page);
+    await expect(mito.getByText('1.00e+0', { exact: true })).toBeVisible();
+
+    await page.keyboard.press('Control+Shift+`');
+    await awaitResponse(page);
+    await expect(mito.getByText('1.00', { exact: true })).toBeVisible();
+  });
+
+  test('Set Datetime Dtype', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await mito.getByTitle('Column1').click();
+
+    await page.keyboard.press('Control+Shift+@');
+    await awaitResponse(page);
+    await expect(mito.getByText('1970-01-01')).toHaveCount(4);
```

Comment:
This test isn't great, since it doesn't catch how the string column can't be turned to a date time. 

Since that is by far the most popular thing that'll happen here, can we:
1. Introduce a new test file called `types.csv` -- one of which is a string column that has date time strings on it.
2. Import this instead of `test.csv` (which we can keep as basic numbers)
3. Test we can change the date correctly for the common case